crates/core/tedge_agent/src/agent.rs

```diff
@@ -146,7 +147,14 @@ impl SmAgentConfig {
         let mqtt_config = mqtt_channel::Config::default()
             .with_host(tedge_config.query(MqttBindAddressSetting)?.to_string())
             .with_port(tedge_config.query(MqttPortSetting)?.into())
-            .with_max_packet_size(10 * 1024 * 1024);
+            .with_max_packet_size(10 * 1024 * 1024)
+            .with_last_will_message(
+                "tedge/health/tedge-agent",
```

Comment:
Don't we have a constant defined for that topic somewhere?

Reply:
Fully agree. Even better than only sharing the topic name, add a `tedge_api::health::get_last_will_message` function to normalize the creation of this last will message.